code/game/objects/structures/crates_lockers/closets/bsdroppod.dm

```diff
@@ -0,0 +1,89 @@
+	allow_dense = TRUE
+	delivery_icon = null
+	can_weld_shut = FALSE
+	armor = list("melee" = 30, "bullet" = 50, "laser" = 50, "energy" = 100, "bomb" = 90, "bio" = 0, "rad" = 0, "fire" = 80, "acid" = 80)
+	anchored = TRUE
+	anchorable = FALSE
+	var/datum/supply_order/SupplyOrder
+
+/obj/structure/closet/bsdroppod/Initialize(mapload, datum/supply_order/so)
+	. = ..()
+	SupplyOrder = so//uses Supply Order passed from expressconsole into BDPtarget
+	addtimer(CALLBACK(src, .proc/open), 30)//open 3seconds after appearing
+
+/obj/structure/closet/bsdroppod/update_icon()//called in initialize, all we have to do is add a door icon
+	cut_overlays()//just incase update_icon somehow gets called multiple times, prevents overlay spam
+	add_overlay("BDP_door")
+
+/obj/structure/closet/bsdroppod/attackby(obj/item/W, mob/user, params)//i no need to be welded or anchored or whatever
+	return
+
+/obj/structure/closet/bsdroppod/toggle(mob/living/user)//ditto
+	return
+
+/obj/structure/closet/bsdroppod/open()
+	var/turf/T = get_turf(src)
+	SupplyOrder.generate(T)//not called during populateContents as supplyorder generation requires a turf
+	cut_overlays()
+	add_overlay("BDP_open")
+	playsound(src, open_sound, 15, 1, -3)
+	QDEL_IN(src, 30)
+
+/obj/structure/closet/bsdroppod/Destroy()//make some sparks b4 deletion
+	do_sparks(10, FALSE, src)
+	QDEL_NULL(SupplyOrder)
+	return ..()
+
+//------------------------------------FALLING BLUESPACE DROP POD-------------------------------------//
+/obj/effect/temp_visual/BDPfall
+	icon = 'icons/obj/2x2.dmi'
+	icon_state = "BDP_falling"
+	pixel_x = -16
+	pixel_y = -5
+	pixel_z = 200
+	name = "Bluespace Drop Pod"
+	desc = "Get out of the way!"
+	layer = FLY_LAYER//that wasnt flying, that was falling with style!
+	randomdir = FALSE
+
+/obj/effect/BDPfall/Initialize()
+	. = ..()//Let's go all the way tonight. No regrets, just love
+
+//------------------------------------TEMPORARY_VISUAL-------------------------------------//
+/obj/effect/BDPtarget
+	icon = 'icons/mob/actions/actions_items.dmi'
+	icon_state = "sniper_zoom"
+	layer = PROJECTILE_HIT_THRESHHOLD_LAYER
+	light_range = 2
+	var/obj/effect/temp_visual/fallingPod
+
+/obj/effect/BDPtarget/Initialize(mapload, datum/supply_order/SO)
+	. = ..()
+	addtimer(CALLBACK(src, .proc/beginLaunch, SO), 30)//wait 3 seconds
+
+/obj/effect/BDPtarget/proc/beginLaunch(datum/supply_order/SO)
+	fallingPod = new /obj/effect/temp_visual/BDPfall(drop_location())
+	animate(fallingPod, pixel_z = 0, time = 3, easing = LINEAR_EASING)//make and animate a falling pod
+	addtimer(CALLBACK(src, .proc/endLaunch, SO, src), 3)//fall 0.3seconds 
+
+/obj/effect/BDPtarget/proc/endLaunch(datum/supply_order/SO)
+	qdel(fallingPod)//delete falling pod after animation's over
+	explosion(src,0,0,2, flame_range = 2) //explosion and camshake
+	for(var/mob/M in oview(7, src))
+		shake_camera(M,2,2)
+	new /obj/structure/closet/bsdroppod(drop_location(), SO)//pod is created after explosion (so that it doesn't get damaged)
```

Comment:
This better be tough enough to survive the explosion because it will get hit by it

Reply:
yeah, figured as much. well it has plenty of explosion armor, so yes